lightning/src/routing/network_graph.rs

```diff
@@ -428,6 +428,70 @@ impl Readable for NodeInfo {
 	}
 }
 
+/// Allows for updating user's network metadata.
+pub trait UpdateMetadata {
+	/// Adds a failed route to track. If we see the route again, we'll reject it or we can take
+	/// some action (like set its fee to make route-finding not find that route again)
+	fn add_failed_route(&mut self, route: Vec<Vec<RouteHop>>);
+	/// If a route starts working again, then we can take it off the failed route list.
+	fn remove_failed_route(&mut self, route: Vec<Vec<RouteHop>>);
```

Comment:
Hmm, I don't think this needs to exist because there isn't really a good way to use it - the trait is really the list of things that rust-lightning will call on implementations (but the struct can have its own impl with its own extra methods users may want). There isn't ever a good criteria for when rust-lightning should decide that a path should be retired (and even if there were, it would mean rust-lightning would need to be track the list of previously-failed routes to tell the UpdateMetadata when to allow it again).

What you may want, instead, is a route_succeeded() call which rust-lightning could use to tell the object that a route passed.